cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -626,6 +640,25 @@ public void onSuccess(V1Status result, int statusCode, Map<String, List<String>>
     return resultFuture;
   }
 
+  /**
+   * Deletes per-run configmap containing config files .
+   */
+  private void deleteConfigMap() {
+    String configMapName = KubeTwillPreparer.CONFIGMAP_NAME_PREFIX + runId.getId();
+    try {
+      coreV1Api.deleteNamespacedConfigMap(configMapName, kubeNamespace, null, null, null, null,
+                                    null, null);
+    } catch (ApiException e) {
+      if (e.getCode() == 404) {
+        LOG.trace("Configmap {} doesn't exist for resource {} of type {}",
+                  configMapName, meta.getName(), resourceType);
+      } else {
+        LOG.warn("Failed to delete configmap {} for resource {} of type {}",
```

Comment:
ApiExceptions should log e.getResponseBody(), as the exception message doesn't normally contain the actual problem.

Reply:
fixed